source/SpinalHDL/Libraries/stream.rst

```diff
@@ -534,6 +534,27 @@ This util take its input stream and routes it to ``outputCount`` stream in a seq
      outputCount = 3
    )
 
+StreamTransactionExtender
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This util helps to extend one input transaction with specified ``count``+1 times of output transactions with the same payload.
+The ``count`` signal would be captured and registered while inputStream fire.
+
+.. code-block:: scala
+
+   val inputStream = Stream(Bits(8 bits))
+   val outputStream = Stream(Bits(8 bits))
+   val count = UInt(3 bits)
+   val extender = StreamTransactionExtender(inputStream, outputStream, count) {
+       (id, payload, last) => payload
```

Comment:
It is not immediately obvious what this line is doing, it also introduces 'id'.  Is that an ordinal index of the count ?  is it 0-based or 1-based, is it 0 to count+1 ?  is `last` from the inputStream's view ?

Maybe a comment will help, or other potential code examples as commented out code will clarify what is possible/allowed/disallowed here.

Reply:
Normally we have 0-based index. ``id`` is just the count value, shows the index of output transfers.

Reply:
Maybe just a comment around it then:

```
   // id, is the 0-based index of total output transfers so far from this input transaction
   // last, is the current inputStream.last indication
   // the returned payload is allowed to be modified
   (id, payload, last) => payload
```

In this context it would be really nice to expose the `repeatIndex` with 0 being the original and 1 being the first repeat, etc...  with it resetting back to zero for the next payload eaten from inputStream, I am assuming it lets you modify the payload in there.

But maybe doing that has a hardware cost when the feature is not used ?  Which would make it a bad idea.